### DISCUSSION TITLE
OKD4.7 offline installation(openstack)

### DISCUSSION THREAD

**Discussion by Mohammed-Idrees** [okd-project/okd#836] [General]
Hi Team,
I'm trying to install okd.47(4.7.0-0.okd-2021-05-22-050008) using mirror registry for offline installation(no internet), I'm able to see bootstrap node is able to pull the images from the mirror registry and when I execute "openshift-install --dir=install_dir/ wait-for bootstrap-complete --log-level=debug" command I'm able to see API is up but the master nodes are not able to pull the images from the mirror registry and I'm not able to ssh to any of the master nodes. is it possible to install OKD4.7 on a cluster without internet?

![image](https://user-images.githubusercontent.com/36164488/130732468-f35cbf7e-1dd2-44e3-a506-c76cbf41b5fc.png)

The console of the master node can be seen like this.

**Comment by JaimeMagiera** [comment]
An I/o timeout indicates there is a network connectivity issue. In this case, the issue is between the control plane node and the bootstrap– not for downloading the images, but attempting to download the node configuration. There are a couple things you might try:
- Verify that network connectivity between the control plane nodes and the bootstrap is unimpeded. 
- increase the timeout for the pull. You can do so by increasing the timeouts:{} in the master.ign
```
"timeouts": {
    	"httpResponseHeaders": 40,
	"httpTotal": 600
    },
```

**Reply by Mohammed-Idrees** [reply]
Hi @JaimeMagiera ,
Thank you for your response. I have tried your solution multiple times even still I'm facing same issue with Master nodes and I have also enabled the required ports as well. 
https://docs.okd.io/latest/installing/installing_bare_metal/installing-bare-metal.html#installation-network-user-infra_installing-bare-metal

**Comment by JaimeMagiera** [comment]
Did you configure the DHCP on your local network appropriately? Do the logs of your DHCP server show that the control plane nodes are requesting an IP? 

https://docs.okd.io/latest/installing/installing_bare_metal/installing-restricted-networks-bare-metal.html#installation-infrastructure-user-infra_installing-restricted-networks-bare-metal

**Reply by Mohammed-Idrees** [reply]
Hi @JaimeMagiera ,
On OpenStack, DHCP server is configured and with the internet I brought clusters and they are working fine. But here I'm trying to bring up a cluster without internet access to the machines(bootstrap, master, and worker). and I'm able to ping all master nodes.

On VMWare we are assigning IPs manually and I'm able to ping to the master node But SSH is not working on that and the console of the Master node on VMWare Dashboard looks like this.
